terraso_backend/apps/core/management/commands/loadbackup.py

```diff
@@ -0,0 +1,275 @@
+                    "AND tablename NOT IN ('core_backgroundtask', 'django_session');"
+                )
+                tables_to_drop = cursor.fetchall()
+                if session_pk:
+                    cursor.execute(
+                        "DELETE FROM django_session WHERE session_key != %s", (str(session_pk),)
+                    )
+                else:
+                    cursor.execute("DELETE FROM django_session")
+                for (table,) in tables_to_drop:
+                    cursor.execute(
+                        sql.SQL("DROP TABLE IF EXISTS {} CASCADE").format(sql.Identifier(table))
+                    )
+            except Exception:
+                msg = "Command failed resetting database"
+                logger.exception(msg)
+                connection.rollback()
+                cleanup()
+                raise CommandError(msg)
+
+            connection.commit()
+
+        try:
+            core_skip = [23, 25]
+            for app_label, version in migrations.items():
+                if app_label == "core":
+                    # need to skip the migration that creates the background tasks
+                    for version in core_skip:
+                        management.call_command(
+                            "migrate", "core", "{0:0>4}".format(version - 1), verbosity=0
+                        )
+                        management.call_command(
+                            "migrate", "core", "{0:0>4}".format(version), fake=True, verbosity=0
+                        )
+                    management.call_command("migrate", "core", verbosity=0)
+                kwargs = {}
+                if app_label == "sessions":
+                    kwargs["fake_initial"] = True
+                management.call_command("migrate", app_label, verbosity=0, **kwargs)
+
+            management.call_command(
+                "loaddata",
+                str(data.resolve()) if isinstance(data, Path) else data,
+                exclude=[
+                    "core.BackgroundTask",
+                    "contenttypes.contenttype",
+                    "auth.Permission",
+                    "sessions.Session",
+                    "admin.LogEntry",
+                ],
+            )
+
+            if user:
+                try:
+                    new_user = User.objects.get(email=user.email)
+                    # provide a fake "email" not in the database
+                    user.email = str(user.id)
+                    user.save()
+                    self._reset_user_id(new_user.id, user.id)
+                    # copy fields from new user to old
+                    for field in User._meta.fields:
+                        if field.name in ("id", "password", "is_staff"):
+                            continue
+                        setattr(user, field.name, getattr(new_user, field.name))
+                    new_user.delete()
+                    user.save()
```

Comment:
This code creates a new user with the same ID as the currently logged in user. This is just to make sure that the user ID of the user triggering the reset does not change and they are not logged out. We copy any other information that might have changed from the user in the DB being restored.